x-pack/plugins/reporting/server/lib/store/store.ts

```diff
@@ -71,19 +72,22 @@ export class ReportingStore {
       return exists;
     }
 
-    const indexSettings = {
-      number_of_shards: 1,
-      auto_expand_replicas: '0-1',
-    };
-    const body = {
-      settings: indexSettings,
-      mappings: {
-        properties: mapping,
-      },
-    };
-
     try {
-      await client.indices.create({ index: indexName, body });
+      await client.indices.create({
+        index: indexName,
+        body: {
+          settings: {
+            number_of_shards: 1,
+            auto_expand_replicas: '0-1',
+            lifecycle: {
+              name: this.ilmPolicyName,
+            },
+          },
+          mappings: {
+            properties: mapping,
+          },
```

Comment:
Moved this in here to take advantage of the TS typed body. If we want to move the body back outside we can also import the payload type to still have TS checking the payload here.